sympy/physics/mechanics/tests/test_actuator.py

```diff
@@ -849,3 +853,144 @@ def test_to_loads(self):
+        self.m, self.g, self.mu_k, self.mu_s, self.v_s, self.sigma, self.F = symbols('m g mu_k mu_s v_s sigma F', real=True)
+
+        # Define the reference frame
+        self.N = ReferenceFrame('N')
+        self.O = Point('O')
+        self.P = self.O.locatenew('P', self.q1 * self.N.x)
+        self.O.set_vel(self.N, 0)
+        self.P.set_vel(self.N, self.u1 * self.N.x)
+        self.pathway = LinearPathway(self.O, self.P)
+
+        self.particle = Particle('particle', self.P, self.m) # Block
+
+        self.system = System(self.N, self.O)
+        self.system.q_ind = [self.q1]
+        self.system.u_ind = [self.u1]
+        self.system.kdes = [self.q1.diff() - self.u1]
+        self.system.add_bodies(self.particle)
+        self.system.apply_uniform_gravity(-self.g * self.N.y)
+        self.system.add_loads(Force(self.particle, self.F * self.N.x))
+
+    def test_block_on_surface_default(self):
+        friction = CoulombKineticFriction(
+            self.mu_k,
+            self.m * self.g,
+            self.pathway,
+        )
+        self.system.add_actuators(friction)
+
+        # Positive velocity case
+        eoms = self.system.form_eoms()
+        expected_positive = self.F - self.g * self.m * self.mu_k - self.m * self.u1.diff()
+        assert (eoms[0] - expected_positive).subs(self.repl_positive) == 0
+
+        # Negative velocity case
+        eoms = self.system.form_eoms()
+        expected_negative = self.F + self.g * self.m * self.mu_k - self.m * self.u1.diff()
+        assert (eoms[0] - expected_negative).subs(self.repl_negative) == 0
+
+    def test_block_on_surface_viscous(self):
+        friction = CoulombKineticFriction(
+            self.mu_k,
+            self.m * self.g,
+            self.pathway,
+            sigma=self.sigma
+        )
+        self.system.add_actuators(friction)
+
+        # Positive velocity case
+        eoms = self.system.form_eoms()
+        expected_positive = self.F - self.g * self.m * self.mu_k + self.sigma * self.u1 - self.m * self.u1.diff()
+        assert (eoms[0] - expected_positive).subs(self.repl_positive) == 0
+
+        # Negative velocity case
+        eoms = self.system.form_eoms()
+        expected_negative = self.F + self.g * self.m * self.mu_k + self.sigma * self.u1 - self.m * self.u1.diff()
+        assert (eoms[0] - expected_negative).subs(self.repl_negative) == 0
+
+    def test_block_on_surface_stribeck(self):
+        friction = CoulombKineticFriction(
+            self.mu_k,
+            self.m * self.g,
+            self.pathway,
+            v_s=self.v_s,
+            mu_s=self.mu_s
+        )
+        self.system.add_actuators(friction)
+
+        # Positive velocity case
+        eoms = self.system.form_eoms()
+        expected_positive = self.F - self.g * self.m * self.mu_k + (-self.g * self.m * self.mu_k + self.g * self.m * self.mu_s) * exp(-self.u1**2/self.v_s**2) - self.m * self.u1.diff()
```

Comment:
```suggestion
        expected_positive = self.F - self.g * self.m * self.mu_k - (self.mu_s - self.mu_s) * self.g * self.m * exp(-self.u1**2/self.v_s**2) - self.m * self.u1.diff()
```
Same here. I expect that this means that there is also a sign error in your implementation.

I would advise to make a simple drawing and write out the equations by hand to check what the answer should be in both the positive and negative case for each of the situations and then check you implementation.

P.S. the force should be equal to `-sign(v) * (f_c + viscous_term + stribeck_term)` if I'm not mistaken.

Reply:
I just wrote down the friction cases by hand and found that there's a sign problem for the viscous and Stribeck terms when the velocity is positive.

Reply:
I think the eoms should be
- Default with viscous term in positive velocity:
   $ma = F - (f_c + \sigma * v)$
- Default with viscous term in negative velocity:
   $ma = F - (-f_c + \sigma * v)$
- Default with Stribeck term in positive velocity:
   $ma = F - (f_c +(f_max - f_c)*e)$
- Default with Stribeck term in negative velocity:
   $ma = F - (-f_c +(f_max - f_c)*e)$

Now the sign of the viscous term works as expected, but the Stribeck term for negative velocity is generated as $+(f_max - f_c)*e$, not $-(f_max - f_c)*e$.

Reply:
I read the paper [Rogner2017] again and thought about this. The friction model is 

$$f_f(v) = f_c * sign(v) + (f_{max} - f_c) * e^(-v/v_s)^2 + \sigma*v$$

When the velocity $v$ is negative, the entire friction force should oppose the motion, and it should have the same overall sign as the default friction term, so the equation should be:

$$f_f(v) = f_c * -sign(v) + (f_{max} - f_c) * e^(-v/v_s)^2 * -sign(v) - \sigma*v$$

which is also discussed in Chapter 4, [Rogner2017].

So in the case of negative velocity, the eom $F + g *m * mu_k + (-g * m * mu_k + g * m * mu_s) * e -m * a$ is valid, and the default friction with the Stribeck term for negative velocity should be:
$ma = F + f_c + (f_{max} - f_c)*e$

So I think current `CoulombKineticFriction` is valid.

Reply:
The signs in the tests indeed seem to be correct. The sign in already in `u1`, which is why the viscous term doesn't require to be multiplied by `sign(u1)`.